airbyte-webapp/src/components/connection/CatalogTree/CatalogTree.tsx

```diff
@@ -56,8 +56,20 @@ const CatalogTreeComponent: React.FC<React.PropsWithChildren<CatalogTreeProps>>
 
   const changedStreams = useMemo(
     () =>
-      streams.filter((stream, idx) => {
-        return stream.config?.selected !== initialValues.syncCatalog.streams[idx].config?.selected;
+      streams.filter((stream) => {
+        if (!stream.config) {
+          return false;
+        }
+        const matchingInitialValue = initialValues.syncCatalog.streams.filter(
+          (initialStream) =>
+            initialStream.stream?.name === stream.stream?.name &&
+            initialStream.stream?.namespace === stream.stream?.namespace
+        )[0];
+        if (!matchingInitialValue) {
+          return false;
+        }
+
+        return stream.config?.selected !== matchingInitialValue.config?.selected;
```

Comment:
Can be checked in the same return statement: `matchingInitialValue?.config?.selected`